sdk/cognitiveservices/ms-azure-cs-inkrecognizer/inkrecognizer/src/main/java/com/azure/ai/inkrecognizer/InkStroke.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package com.azure.ai.inkrecognizer;
+
+/**
+ * The InkStroke interface represents an ink stroke (a collection of ink points from the time a user places his writing instrument on
+ * the writing surface until the the instrument is lifted.
+ * Clients of the Ink Recognizer services are expected to implement this interface on the data store
+ * used to store ink data so the InkRecognizer object can use it to translate the ink to JSON for delivery
+ * to the Ink Recognizer service.
+ * @author Microsoft
+ * @version 1.0
+ */
+public interface InkStroke {
+
+    /**
+     * Retrieves the points contained in the strokes.
+     * @return A list of points in the stroke.
+     */
+    Iterable<InkPoint> getInkPoints();
+
+    /**
+     * Retrieves the InkStrokeKind. The default is InkStrokeKind.UNKNOWN.
```

Comment:
If the store returns UNKNOWN on strokes, and the AppKind is set to "Drawing," will the app kind override the stroke kind? I.e. does "UNKNOWN" == "not set"?

Reply:
The stroke kind is sent to the server only if it set by the client, hence the only way to set stroke kind as UNKNOWN is to be set as UNKNWON by the client directly. If the stroke kind is set as UNKNOWN by the client and the AppKind is set to 'Drawing', both of these pieces of information are provided to the server for decisions about overrides.

Reply:
correct. The server does the disambiguation. In the case, the kind on the stroke wins.

Reply:
Code has been modified to treat InkStrokeKind.UNKNOWN same as when the app client does not set the InkStrokeKind. InkStrokeKind is not sent to the server in these two cases as no meaningful action can be taken with this information.